### ISSUE TITLE
Excluded regex URL patterns are ignored by Active Scanner on first run since application start

### ISSUE THREAD

**Issue by icongit4c** [zaproxy/zaproxy#6476]
**Describe the bug**
During the first run of Active Scan after opening the ZAP application, requests are made to URLs that should be out of scope.  These are URLs that match the exclude regex specified for the Scanner (Session->Exclude from scanner).
On subsequent runs of the Active Scanner (without restarting ZAP after the first run), the Active Scanner works fine.  Requests are only made to URLs within the site that was added to the context (Session->Contexts->Default Context->Include in context) and
exclude URL regex patterns are not ignored.

**To Reproduce**

Re-open ZAP if you have run an Active Scan since opening ZAP.
In a new session:
1.)Session->Contexts->Default Context->Include in context: https:\/\/yoursite\.com.*
2.)Session->Exclude from proxy: .*mozilla\.com.*
3.) Run active scan with
Starting point: yoursite
Policy: Default policy
Context: Default context

In the Active Scan log, you will see requests are still sent to mozilla.com.  mozilla.com is just an example.  There are other URLs that I have excluded, but still show in the Active Scan logs.

**Expected behavior**
The Active Scan should not ignore the regex URL patterns specified for the scanner on the first run.

**Screenshots**

Problem example:
<img width="848" alt="Problem example" src="https://user-images.githubusercontent.com/80282663/110369341-46c7c600-8018-11eb-9e4d-1334090aad72.png">

First scan (1215 requests):
<img width="943" alt="First scan - 1215 requests" src="https://user-images.githubusercontent.com/80282663/110369414-5f37e080-8018-11eb-8a7d-7c6405da4ced.png">

Second scan (435 requests):
<img width="937" alt="Second scan - 435 requests" src="https://user-images.githubusercontent.com/80282663/110369448-69f27580-8018-11eb-8568-b3cdacae34d9.png">

**Software versions**
 - ZAP: 2.10.0
 - Add-on: None
 - OS: Windows 10
 - Java: oracle java 11.0.5
 - Browser: Firefox 73.0,  Chrome 89.0.4389.82 (Official Build) (64-bit)

**Errors from the zap.log file**
There are no errors in the log from the Active Scan run.

**Additional context**
The Active Scan doesn't start using the excluded domains until the 87% progress mark.  This is consistent on every run.  Until the 87% mark, only my domain's URLs show in the
Active Scan log.  At 87%, the scan progress seems to freeze for ~10 seconds and I see a file save/open dialog quickly box open and close on it's own.  After that point, the Active Scan
continues and I start to see the domains that should be excluded.

If I run a second scan, the scan works properly.  Only URLs for my domain show in the active scan log.  The problem is repeatable only on the first run of the Active Scan after the
application is started.  On subsequent runs, the Active Scan works fine and the number of requests reported is smaller than the first run.  You have to close/reopen the ZAP application, load the session file and run the Active Scan again to
reproduce the problem.

**Comment by thc202** [comment]
Are you using the DOM XSS scan rule?

**Comment by thc202** [comment]
Those URLs are accessed by Firefox (used by the DOM XSS scan rule), they are not actually being scanned though. Still a nuisance, we do try to disable those kind of requests... but not always possible.

**Comment by icongit4c** [comment]
It does seem like the add-on is modifying some of the requests to the URLs that should be ignored.  Here's an example.

Outside of ZAP, if I navigate to our site and look at the browser network trace, I can see this GET request is made only once when loading our site.  This domain is not ours, it is for a 3rd party
product we're using.
https://uap.sencha.com/ping.gif?u=https%3A%2F%2Fmysite%3A8443%2F&p=gxt&v=gxt%204.0.3

The URL always has the same parameters and our site only calls it on load.
But, in the Attack Scanner logs (first run of the scanner), I see this form of the GET request, which has been modified.  The request was sent and received a response from the 3rd party domain.

GET https://uap.sencha.com/ping.gif?u=https%3A%2F%2Fmysite%3A8443%2F%23jaVasCript%3A%2F*-%2F*%2560%2F*%5C%2560%2F*%27%2F*%2522%2F**%2F(%2F*%2520*%2FoNcliCk%3Dalert()%2520)%2F%2F%250D%250A%250d%250a%2F%2F%253C%2FstYle%2F%253C%2FtitLe%2F%253C%2FteXtarEa%2F%253C%2FscRipt%2F--!%253E%5Cx3csVg%2F%253CsVg%2FoNloAd%3Dalert()%2F%2F%253E%5Cx3e&p=gxt&v=gxt%204.0.3 HTTP/1.1
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:73.0) Gecko/20100101 Firefox/73.0
Accept: image/webp,*/*
Accept-Language: en-US,en;q=0.5
Connection: keep-alive
Referer: https://mysite
Content-Length: 0
Host: uap.sencha.com

I've added the Sencha domain to the scanner exclusion regex list, but I still see it on the first run and it looks like the request is being modified and set.

**Comment by thc202** [comment]
It's the target site that's being modified/scanned which then does that request, if we block those requests we could be missing vulnerabilities.

**Comment by icongit4c** [comment]
Thanks for recommendation. We'll try it out.
